### PR TITLE
Support I/O mapping variables starting with _

### LINKED ISSUE #11397: Support I/O mapping variables starting with _
**Is your feature request related to a problem? Please describe.**
At the moment it is not possible to use I/O mappings where the variable name starts with an `_`.  It is possible to create variable names like this in other ways, for example using the `SetVariables` RPC.

Allowing variables starting with an `_` allows users to "group" a specific type of variable. E.g. internal variables for Connectors.

At the moment it is not possible to deploy a model that contains I/O mapping variables names starting with `_`. The deployment gets rejected:
```
Command 'CREATE' rejected with code 'INVALID_ARGUMENT': Expected to deploy new resources, but encountered the following errors:
'diagram_2.bpmn': - Element: StartEvent_1 > extensionElements > ioMapping > output
    - ERROR: Expected path expression '_variable' but doesn't match the pattern '[a-zA-Z][a-zA-Z0-9_]*(\.[a-zA-Z][a-zA-Z0-9_]*)*'.
 [ deploy-error ]
```
The validation happens in the [ZeebeExpressionValidator](https://github.com/camunda/zeebe/blob/main/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/validation/ZeebeExpressionValidator.java#L26-L27)

**Describe the solution you'd like**
Restrictions around variable names should be consistent across the system. If we allow variables to start with an `_` in one place, we should also allow it in the I/O mappings.

**Describe alternatives you've considered**
Instead of allowing it in I/O mappings we could disallow it in other places. This has a much bigger impact and makes it so we'd have to think about running processes that already contain these variables.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by skayliu**
## Description

At the moment it is not possible to use I/O mappings where the variable name starts with an _. It is possible to create variable names like this in other ways, for example using the SetVariables RPC.

Restrictions around variable names should be consistent across the system. If we allow variables to start with an _ in one place, we should also allow it in the I/O mappings.

## Related issues

closes #11397 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.ink/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [x] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.ink/camunda/operate/issues)
- [ ] [Tasklist](https://github.ink/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.ink/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.ink/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.ink/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.ink/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by skayliu** [comment]
Hey, @korthout, ✋  Please take a moment to review this PR when you have time. Thank you.

**Review by nicpuppa — COMMENTED** [review]
Hey @skayliu thanks for the great work. Overall I don't see any issue with your implementation. I have just a suggestion/question.

🔧 Should we add more tests in other places ? (eg. JobInputMappingTest or SignalEndEventOutputMappingTest ecc.) 

There are more test classes that use variables, what do you think ?

**Comment by skayliu** [comment]
> Hey @skayliu thanks for the great work. Overall I don't see any issue with your implementation. I have just a suggestion/question.
> 
> 🔧 Should we add more tests in other places ? (eg. JobInputMappingTest or SignalEndEventOutputMappingTest ecc.)
> 
> There are more test classes that use variables, what do you think ?

Ok, I'll add for those tests.

**Review by nicpuppa — CHANGES_REQUESTED** [review]
Thanks @skayliu for the great work 🚀 

I added some more question/suggestion. Please let me what you thought 🍪

File: engine/src/test/java/io/camunda/zeebe/engine/processing/variable/mapping/MessageCatchElementOutputMappingTest.java [hunk 1]
```diff
@@ -44,7 +44,7 @@ public final class MessageCatchElementOutputMappingTest {
   @ClassRule public static final EngineRule ENGINE_RULE = EngineRule.singlePartition();
   private static final String PROCESS_ID = "process";
   private static final String MESSAGE_NAME = "message";
-  private static final String CORRELATION_VARIABLE = "key";
+  private static final String CORRELATION_VARIABLE = "_key";
```
[the remark this answers is not in the document]
Comment (skayliu) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/variable/mapping/MessageCatchElementOutputMappingTest.java:47]: ok, i‘ll revert this.

File: engine/src/test/java/io/camunda/zeebe/engine/processing/variable/mapping/MessageCatchElementOutputMappingTest.java [hunk 2]
```diff
@@ -295,7 +295,7 @@ private long triggerProcessInstance(final Record<DeploymentRecordValue> deployme
         .message()
         .withCorrelationKey(correlationKey)
         .withName(MESSAGE_NAME)
-        .withVariables(asMsgPack("foo", "bar"))
+        .withVariables(asMsgPack("_foo", "bar"))
```
[the remark this answers is not in the document]
Comment (skayliu) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/variable/mapping/MessageCatchElementOutputMappingTest.java]: I've applied you suggestion, please review again.

**Review by nicpuppa — APPROVED** [review]
Thanks @skayliu , LGTM. Great work! 🚀 

Can you please rebase your branch ? Thank you

**Comment by skayliu** [comment]
> Thanks @skayliu , LGTM. Great work! 🚀
> 
> Can you please rebase your branch ? Thank you

It's done, @nicpuppa, should i squash these commits now?

**Comment by nicpuppa** [comment]
> It's done, @nicpuppa, should i squash these commits now?

Yes please 👍

**Review by nicpuppa — APPROVED** [review]

**Comment by nicpuppa** [comment]
Hey @skayliu can you check the commit linter ?

**Pull request merged**